File: ts/src/vector/flat/longFlatVector.ts
import { FixedSizeVector } from "../fixedSizeVector";

export class LongFlatVector extends FixedSizeVector<BigInt64Array | BigUint64Array, bigint> {
    protected getValueFromBuffer(index: number): bigint {
        return this.dataBuffer[index];
    }
}

File: java/src/main/java/org/maplibre/mlt/vector/flat/LongFlatVector.java
package org.maplibre.mlt.vector.flat;

import java.nio.LongBuffer;
import org.maplibre.mlt.vector.BitVector;
import org.maplibre.mlt.vector.Vector;

public class LongFlatVector extends Vector<LongBuffer, Long> {
  public LongFlatVector(String name, LongBuffer dataBuffer, int size) {
    super(name, dataBuffer, size);
  }

  public LongFlatVector(String name, BitVector nullabilityBuffer, LongBuffer dataBuffer) {
    super(name, nullabilityBuffer, dataBuffer);
  }

  @Override
  protected Long getValueFromBuffer(int index) {
    return dataBuffer.get(index);
  }
}
